### PR TITLE
Add required repo files

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Also:
* rename migration.md to MIGRATING.md
* include post-release instructions in RELEASING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
```
Comment (jkwatson) [inline CHANGELOG.md:8]: can we pre-populate this with the last couple of releases also?

Comment (mateuszrzeszutek) [reply CHANGELOG.md:8]: 👍 

Will do that -- I think last 2-3 releases are enough

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,4 +7,31 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Added
+
+- Instrument Netty 4.0 to add Server-Timing header [#108](https://github.com/signalfx/splunk-otel-java/pull/108)
+
+### Changed
+
+- Use upstream opentelemetry-java-instrumentation 0.16.1. [#117](https://github.com/signalfx/splunk-otel-java/pull/117)
+
 [Unreleased]: https://github.com/signalfx/splunk-otel-java/compare/v0.9.0...HEAD
```
Comment (jkwatson) [inline CHANGELOG.md:35]: should this be a proper link, and up in the Unreleased section, rather than down here hidden?

Comment (mateuszrzeszutek) [reply CHANGELOG.md:35]: I used the same pattern as described in https://keepachangelog.com/en/1.1.0/

I suppose it's easier to manage multiple links this way once you have lots of them.

**Review by jkwatson — APPROVED** [review]

**Pull request merged**